Tests/SonarQube.Bootstrapper.Tests/ArgumentProcessorTests.cs

```diff
@@ -257,9 +256,72 @@ public void ArgProc_BeginAndEndVerbs()
             logger.AssertSingleErrorExists("begin", "end");
         }
 
-        #endregion
+        [TestMethod]
+        public void ArgProc_Verbosity()
+        {
+            CheckVerbosity(VerbosityCalculator.DefaultLoggingVerbosity, "Default verbosity does not match");
+            CheckVerbosity(VerbosityCalculator.DefaultLoggingVerbosity, "Settings are case sensitive", "/d:sonar.VERBOSE=true");
+            CheckVerbosity(VerbosityCalculator.DefaultLoggingVerbosity, "Trace and Debug are independent SonarQube verbosity settings. We only track Debug", "/d:sonar.log.level=trace");
+
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.verbose=true");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.verbose=TRue");
+            CheckVerbosity(LoggerVerbosity.Info, "", "/d:sonar.verbose=false");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=debug");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=INFO|DEBUG|TRACE");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=***debug***");
+            CheckVerbosity(LoggerVerbosity.Debug, "sonar.verbose takes precedence over sonar.log.level", "/d:sonar.verbose=true", "/d:sonar.log.level=INFO");
```

Comment:
A statement about the expected order of precedence would be useful in the relevant point in the source code to make it clear that ordering is deliberate.